python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -745,10 +751,13 @@ def write(self, value: Any, timeout: Optional[float] = None) -> None:
         for channel in self._channels:
             channel.write(value, timeout)
 
-    def read(self, timeout: Optional[float] = None) -> Any:
+    def read(
+        self, actor_id: Optional[str] = None, timeout: Optional[float] = None
+    ) -> Any:
         self.ensure_registered_as_reader()
-        actor_id = self._get_self_actor_id()
-        return self._channel_dict[actor_id].read(timeout)
+        if actor_id is None:
+            actor_id = self._get_self_actor_id()
```

Comment:
wondering if we can get rid of this old logic entirely, is there any situation where read would be called from a task node that wasn't an actor?

Reply:
Why would we want to get rid of `_get_self_actor_id()`? We could use it to determine whether it is an actor or the driver, and then decide which channel to use.

Reply:
we're passing the actor_id to read now, so i don't think the actor_id should be None in any situation, because readers are only created in executable tasks or the driver

executable tasks always have an actor, and driver always has proxy actor

Reply:
will try removing and seeing if all tests still pass